src/library/dao/playlistdao.cpp

```diff
@@ -227,6 +228,51 @@ void PlaylistDAO::deletePlaylist(const int playlistId) {
     }
 }
 
+int PlaylistDAO::deleteAllPlaylistsWithFewerItems(PlaylistDAO::HiddenType type, int length) {
+    QSqlQuery query(m_database);
+
+    ScopedTransaction transaction(m_database);
```

Comment:
Move the start of the transaction to the top.

As already mentioned this should be extracted from the DAOs. All DAOs should safely assume that their functions are executed in a proper transaction scope. They should not (need) to make any assumptions about the outer context.

Reply:
Just noticed that you added this transaction recently:
- Move the transaction out of PlaylistDAO and to SetlogFeature 
- Add a comment to the function docs that states that it has to be executed within a transaction.

Unfortunately, we cannot determine if a transaction has already been started. This would require to introduce a wrapper for QSqlDatabase.